optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/plan/factories/UpgradePlanFactory.java

```diff
@@ -7,9 +7,42 @@
  */
 package io.camunda.optimize.upgrade.plan.factories;
 
+import io.camunda.optimize.service.db.DatabaseClient;
+import io.camunda.optimize.service.db.schema.OptimizeIndexNameService;
+import io.camunda.optimize.upgrade.exception.UpgradeRuntimeException;
 import io.camunda.optimize.upgrade.plan.UpgradeExecutionDependencies;
 import io.camunda.optimize.upgrade.plan.UpgradePlan;
+import java.io.IOException;
 
 public interface UpgradePlanFactory {
   UpgradePlan createUpgradePlan(final UpgradeExecutionDependencies upgradeExecutionDependencies);
+
+  void logErrorMessage(final String message);
+
+  default boolean isC7LicenseDataPresent(
+      final DatabaseClient databaseClient, final OptimizeIndexNameService indexNameService) {
+    try {
+      return databaseClient.countWithoutPrefixWithExistsCheck(
+              indexNameService.getOptimizeIndexAliasForIndex("license"))
+          > 0;
+    } catch (IOException e) {
+      final String reason = "Was not able to determine existence of C7 data.";
+      logErrorMessage(reason);
+      throw new UpgradeRuntimeException(reason, e);
+    }
+  }
+
+  default void validateOptimizeModeAndFailIfC7DataPresent(
+      final DatabaseClient databaseClient, final OptimizeIndexNameService indexNameService) {
+    if (isC7LicenseDataPresent(databaseClient, indexNameService)) {
```

Comment:
Note that some users might have been passing a license via an env var or file. They would not be covered in this case 🤔 

Along with good documentation, this might be enough for now

Reply:
If they are using a file then the first time this is read it is also stored in the license index, so should be fine as long as the upgrade isnt run before Optimize is starteda aat least once. I am not familiar with using an env var for the license on C7 and dont see that in the config, license manager or the docs 🧐 do we not store it in that case?